### PR TITLE
[tree/RDF] Move GTLBN from RDF to TreeInternalUtils

### PR DESCRIPTION
**Pull request by dkpoult**
# This Pull request:

Moves GetTopLevelBranchNames from the RDF Internals namespace to the Internal TreeUtils namespace, as it does not rely on RDF and creates unnecessary dependencies.

## Changes or fixes:
- Move GetTopLevelBranch names and its implementation from `/tree/dataframe` to `/tree/tree`.

## Checklist:

- [X] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes N/A

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx [hunk 1]
```diff
@@ -663,7 +662,7 @@ struct RNeedJittingHelper<RInferredType> {
    static constexpr bool value = true;
 };
 
-template <typename ...ColTypes>
+template <typename... ColTypes>
```
Comment (eguiraud) [inline tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx:665]: there is a number of unrelated whitespace changes in the commit, can you please remove them?

(`clang-format FILENAME` will reformat the whole file, `git clang-format --diff master HEAD | git apply` will reformat only lines around those the commits in range `master..HEAD` touched)

File: tree/tree/src/InternalTreeUtils.cxx [hunk 2]
```diff
@@ -18,6 +18,45 @@
...
+   auto branches = t.GetListOfBranches();
+   if (branches) {
+      for (auto branchObj : *branches) {
+         const auto name = branchObj->GetName();
+         if (bNamesReg.insert(name).second) {
+            bNames.emplace_back(name);
+         } else if (!friendName.empty()) {
+            // If this is a friend and the branch name has already been inserted, it might be because the friend
+            // has a branch with the same name as a branch in the main tree. Let's add it as <friendname>.<branchname>.
+            // If used for a Snapshot, this name will become <friendname>_<branchname> (with an underscore).
```
Comment (eguiraud) [inline tree/tree/src/InternalTreeUtils.cxx:40]: this line about Snapshot is not relevant anymore

File: tree/tree/src/InternalTreeUtils.cxx [hunk 3]
```diff
@@ -78,6 +117,17 @@ void RFriendInfo::AddFriend(const std::vector<std::pair<std::string, std::string
    }
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// Get all the top-level branches names, including the ones of the friend trees
+std::vector<std::string> GetTopLevelBranchNames(TTree &t)
+{
+   std::set<std::string> bNamesSet;
```
Comment (eguiraud) [inline tree/tree/src/InternalTreeUtils.cxx:124]: I know this was already there, but I think the safe default, for performance in case of a large number of branches, is `std::unordered_set` rather than `std::set`.

**Review by eguiraud — CHANGES_REQUESTED** [review]
looks good but too many unrelated whitespace changes :)

File: tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx (hunk 1 above)
Comment (dkpoult) [reply tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx:665]: Okay noted - I should have asked which would have been better, sorted now!

Comment (dkpoult) [reply tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx]: Excess whitespace is no more :)

Comment (eguiraud) [reply tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx]: I would have been ok with the unrelated whitespace changes in a separate commit -- but it's also ok to leave them as they are

**Review by eguiraud — APPROVED** [review]
LGTM, thanks!
